File: Problem Set 2/Caesar/caesar.c
/* Program to encrypt a message using the Caesar cipher */

#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>
#include <stdio.h>

int main(int argc, string argv[])
{
    string argument = argv[1];
    int key;
    
    // Check key from command line argument
    while (argc != 2)
    {
        printf("Usage: %s key\n", argv[0]);   
        return 1;
    }
    
    // Check command line argument
    for (int i = 0; i < strlen(argument); i++)
    {
        // Check argument to make sure all characters are digits
        if (isalpha(argument[i]))
        {
            printf("Usage: %s key\n", argv[0]);   
            return 1;
        }
        else
        {
            // Turn key into integer
            key = atoi(argument); 
        }
    }
        
    // Prompt user for plaintext
    string plaintext = get_string("plaintext: ");
    
    // Encipher each character in the plaintext string
    for (int i = 0; i < strlen(plaintext); i++)
    {
        // If character is an uppercase character
        if (isupper(plaintext[i]))
        {
            plaintext[i] = (((plaintext[i] + key - 65) % 26) + 65);
        }
        // If character is a lowercase character
        if (islower(plaintext[i]))
        {
            plaintext[i] = (((plaintext[i] + key - 97) % 26) + 97);
        }
        // If character is symbol character
        if (ispunct(plaintext[i]))
        {
            plaintext[i] = plaintext[i];
        }
    }
    
    // Print ciphertext
    printf("ciphertext: %s\n", plaintext);
}

File: Problem Set 6/Caesar/caesar.py
# Program to encrypt a message using the Caesar cipher 

import sys 
from cs50 import get_string

def main():
    # Check key from command line argument
    while len(sys.argv) != 2:
        sys.exit("Usage: python sys.argv[1] key")
        
    # Check command line argument
    for c in sys.argv[1]:
        # Check argument to make sure all characters are digits
        if c.isalpha():
            sys.exit("Usage: python sys.argv[1] key")
    
    # Turn key into integer
    key = int(sys.argv[1])
    
    # Prompt user for plaintext
    plaintext = get_string("plaintext: ")
    
    # Create a list for ciphertext characters
    ciphertext = []
    
    # Encipher each character in the plaintext string
    for c in plaintext:
        if c.isalpha():
            ciphertext.append(cipher(c, key))
        else:
            ciphertext.append(c)
    
    # Print ciphertext
    print("ciphertext: ", end='')
    print("".join(ciphertext))

def cipher(c, key):
    if c.isupper():
        return chr(((ord(c) - 65 + key) % 26) + 65)
    else:
        return chr(((ord(c) - 97 + key) % 26) + 97)

if __name__ == "__main__":
    main()
